_posts/2020-12-16-#19935.md

```diff
@@ -7,17 +7,80 @@ authors: [achow101]
 components: ["refactoring"]
 host: willcl-ark
 status: upcoming
-commit:
+commit: 19935
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+The PR is mostly moving existing code into two new files:
+```
+src/util/hasher.{cpp|h}
+```
+
+But touches some interesting code in:
+```
+src/coins.{cpp|h}
+src/index/blockfilter.h
+src/script/sigcache.h
+src/txmempool.{cpp|h}
+src/validation.h
+```
+
+Hashers are generally used to construct hashmaps of groups of items for fast item lookup, addition and removal.
+
+They allow lookup of elements in `0(1)` lookup time in the typical/best-case scenario although implementation can affect this. You can see the implementation of `std::unordered_map` here: [cppreference/unordered_map](https://en.cppreference.com/w/cpp/container/unordered_map). For even more detail, take a look at this blog post: [Hash functions for C++ Unordered Containers](https://marknelson.us/posts/2011/09/03/hash-functions-for-c-unordered-containers.html).
```

Comment:
"They allow lookup of elements in `0(1)` lookup time"

If you retouch, can drop the second "lookup" here

Reply:
done